miio/gateway.py

```diff
@@ -145,14 +155,22 @@ def discover_devices(self):
         self._devices = []
 
         for x in range(0, len(devices_raw), 5):
+            # Extract discovered information
+            dev_info = SubDeviceInfo
```

Comment:
Simply use `SubDeviceInfo(*devices_raw[x : x + 5])` here, it should work just fine :-)

Reply:
nice